### ISSUE TITLE
Not queueing images

### ISSUE THREAD

**Issue by sgtpenguin** [dispositiontools/craft-alt-text-generator#6]
Hey I've purchased and started trying out the plugin today and around 80 or so images have been processed.  Some did not error but still failed to retrieve and update alt text and needed to be manually resubmitted.  But the main issue I'm having now is that those have been processed but there are around 4,000 images on the site and most if not all do not have alt text.  But when I click the utility no more images get added to the queue.

**Comment by dispositiontools** [comment]
Hi, I’m sorry about this - is it possible for you to share the alt text generator database tables with me? At support@disposition.tools - if it’s just those tables then there shouldn’t be any personal data in there except the image urls. If that’s ok that will help me see the issue.  

Were there any queue errors?

**Comment by sgtpenguin** [comment]
No queue errors.  Sending `alt-text-generator_alttextaiapicalls`.  Thats the only table I see.  Thanks for taking a look!

**Comment by dispositiontools** [comment]
Hi, thanks for sending the file. I've taken a lot and it's not making a lot of sense yet. Would you be up for a screen share to investigate? 

One thing to ask is about the rest of the assets. 
The plugin currently only queues them if they are less than 10mb, over 50x50 and if they are in ['jpg', 'gif', 'png', 'webp', 'jpeg'], and if they are externally accessible 

I'm working on a fix for all this.

**Comment by sgtpenguin** [comment]
There are for sure some assets over 10mb but not that many.  Almost all, or all are over 50x50 and are mostly the acceptable formats (there's definitely some svg, mp4, etc mixed in there).  They are public urls and images that are in use on the site.

I put together a little `test.php`  script in my `/web` folder to put some numbers out of this part of the process.  Here are the results.
<img width="677" alt="Screenshot 2024-06-19 at 8 05 39 AM" src="https://github.com/dispositiontools/craft-alt-text-generator/assets/16600959/8b8c5967-add0-409a-9335-b3b3a6bbd501">

Here is the script
```<?php
require dirname(__DIR__) . '/bootstrap.php';

// Load and run Craft
/** @var craft\web\Application $app */
$app = require CRAFT_VENDOR_PATH . '/craftcms/cms/bootstrap/web.php';
use craft\elements\Asset as AssetElement;
use dispositiontools\craftalttextgenerator\AltTextGenerator;

$assetsQuery = AssetElement::find()->kind('image')->hasAlt(false);
$assets = $assetsQuery->all();

echo 'Credits Avail: '.  AltTextGenerator::getInstance()->altTextAiApi->getNumberOfAltTextApiCredits() . '<br />';
echo 'Total Images: ' . count($assets) . '<br />';
$suitableCount = 0;
foreach ($assets as $asset) {
    $suitability = AltTextGenerator::getInstance()->altTextAiApi->checkAssetSuitability($asset);
    if ($suitability['success']) {
        $suitableCount++;
    }
}
echo 'Suitable Images: ' . $suitableCount;
?>
```

So I didn't really continue tracing from there but this indicates that I don't think the asset query, suitability test, or token limit are to blame here.  It seems like maybe it's hitting a dead end conditional somewhere in the request process that doesn't report an error and also doesn't continue the loop.  Considering that a whole bunch went through and now no new images get queued (unless they are newly uploaded, that appears to be working), that led me to think maybe it was one image in particular, but to test that theory I manually added alt text to the first passing image so that it would be skipped, thinking maybe another batch would go through, but still no additional images are processed

Noting also Craft has nearly 4k images indexed, but almost 10k come back in the query.  Definitely something I need to consider.
<img width="498" alt="Screenshot 2024-06-19 at 8 07 22 AM" src="https://github.com/dispositiontools/craft-alt-text-generator/assets/16600959/65ea4b21-51d0-4dd9-8d9c-e3996f7a6371">

Yea I'd be happy to screen share and talk through it.  Feel free to reach me over email at dylan@hyperakt.com
Thanks

**Comment by dispositiontools** [comment]
I've pushed an update which adds some logging, but also generates a report when you use the Queue all utility which should help us diagnose the first step - ie if it's adding the assets to the queue. And I've added some logging to the queue items. 

I've also had to remove the human review as alttext.ai remove that option from the api tomorrow. 

I've also added the ability to resubmit an asset to alttext.ai when using the queue all utility. Which will be worth trying to force alttext.ai to start the process again with all images that it might have had a problem with previously.

**Comment by sgtpenguin** [comment]
Fantastic! @dispositiontools   I was able to spot the problem.  In `services/AltTextApi.php` on line `926` `$requestCount` is incremented before asset suitability is being determined.  `$requestCount` should only increment when an asset is suitable.

As such in my case I have 40 credits remaining.  Since I have looped through 40 unsuitable images, the plugin thinks I have no credits left even though none have been used.

**Comment by dispositiontools** [comment]
ah. I'm so sorry, and thank you.  I'll get that fixed right now. There will be a new release this evening.

**Comment by dispositiontools** [comment]
I've pushed versions v4.1.1 and v5.1.1 which fixes this issue. 

I really appreciate you taking the time to help.
